### PR TITLE
Feature 40 support for more complex video controls

### LINKED ISSUE #40: Support for more complex video controls

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #40

### CODE REVIEW

File: lib/widgets/noheva_video.dart [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+          key: key,
+          element: element,
+          children: children,
+        );
+
+  @override
+  NohevaVideoState createState() => NohevaVideoState();
+}
+
+class NohevaVideoState extends NohevaWidgetState<NohevaVideo> {
```
Comment (anttileppa) [inline lib/widgets/noheva_video.dart:21]: Add docs

File: lib/widgets/noheva_widget.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  const NohevaWidget({
+    Key? key,
+    required this.element,
+    this.hidden,
+    this.onTap,
+    this.children = const [],
+  }) : super(key: key);
+}
+
+abstract class NohevaWidgetState<T extends NohevaWidget> extends State<T> {
```
Comment (anttileppa) [inline lib/widgets/noheva_widget.dart:26]: Add docs

**Review by anttileppa — APPROVED** [review]

File: lib/widgets/noheva_video.dart (hunk 1 above)
Comment (villejuutila) [reply lib/widgets/noheva_video.dart:21]: Resolved.

File: lib/widgets/noheva_widget.dart (hunk 2 above)
Comment (villejuutila) [reply lib/widgets/noheva_widget.dart:26]: Resolved.

**Pull request merged**